src/syntax/node.rs

```diff
@@ -138,14 +138,23 @@ impl SyntaxNode {
         }
     }
 
+    pub fn err_with_hint(&mut self, hint: impl Into<EcoString>) {
+        match &mut self.0 {
+            Repr::Error(error) => Arc::make_mut(error).with_hint(hint),
```

Comment:
I am not sure if `make_mut` is correct here. It is used somewhere else in the file however.

Reply:
It's fine.